apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.SignedPostPolicy.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright 2020 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Google.Cloud.Storage.V1
+{
+    public sealed partial class UrlSigner
+    {
+        /// <summary>
+        /// The result of signing a <see cref="PostPolicy"/>
+        /// containing the fields that should be including when
+        /// making the post request.
+        /// </summary>
+        public sealed class SignedPostPolicy
+        {
+            private const string SignatureElementName = "x-goog-signature";
+            private const string PolicyElementName = "policy";
+
+            internal SignedPostPolicy(PostPolicy source, string encodedPolicy, string signature, DateTimeOffset expiration, Uri url)
+            {
+                Source = GaxPreconditions.CheckNotNull(source, nameof(source));
```

Comment:
Should we clone the source here? Currently the user could create a policy, get a signed version, add a field, and then the signed policy becomes invalid.